virttest/qemu_devices/qcontainer.py

```diff
@@ -3091,3 +3093,82 @@ def _emulator_setup(binary, extra_options=None):
         devices.append(tpm_model_dev)
 
         return devices
+
+    def numa_hmat_lb_define_by_params(self, nodeid, initiator, params):
+        """
+        Create numa hmat-lb device by params
+
+        :param nodeid: The target node id of the hmat-lb device
+        :param initiator: The initiator of the hmat-lb device
+        :param params: The params for hmat-lb
+        :return: The qdevices.QCustomDevice of hmat-lb
+        """
+        aobject = '%s_hmat_lb' % nodeid
+        hmat_lb_params = {'target': nodeid, 'initiator': initiator,
+                          'hierarchy': params['numa_hmat_lb_hierarchy'],
+                          'hmat_type': 'hmat-lb'}
+        data_type = params['numa_hmat_lb_data_type']
+        hmat_lb_params.update({'data-type': data_type})
+        if 'latency' in data_type:
+            hmat_lb_params.update(
+                {'latency': params['numa_hmat_lb_latency']})
+            aobject += '_latency'
+        elif 'bandwidth' in data_type:
+            hmat_lb_params.update(
+                {'bandwidth': params['numa_hmat_lb_bandwidth']})
+            aobject += '_bandwidth'
+        return qdevices.QCustomDevice('numa', params=hmat_lb_params,
```

Comment:
I'm not sure if it's suitable to define it as a kind of qdevices.QCustomDevice though it can work.

Reply:
@huangyum Well, then a `QStringDevice`? It is still a "device" in the avocado-vt context, I think that's a different concept comparing with the qemu-kvm context. From my perspective those classes are just data structures to describe certain object, I think `-numa hmat-lb` + `xx=xx,xx=xx` pattern is very suitable for `QCustomDevice` definition. And comparing to `QStringDevice` it manages objects with proper properties, which is more object oriented I think.

Reply:
@huangyum Having said that, if you prefer `QStringDevice` or other object I would like to modify it, let me know your thought, thanks.

Reply:
I agree, the description of QCustomDevice seems suit here. "Representation of the '-$option $param1=$value1,$param2...' qemu object."
Actually I was bothered by the 'Device'. But you've made a point that this is  avocado-vt context, so I'm fine with it. Thanks.